File: users/user_controller.py
from fastapi import APIRouter, Depends
from users import user_service as userService
from common.database.db import get_db
from users.dtos.create_user import CreateUserInput
from sqlalchemy.orm import Session
from users.dtos.read_users import ReadUsersInput
from users.dtos.update_user import UpdateUserInput
from users.dtos.delete_user import DeleteUserInput

app = APIRouter(
    prefix="/user",
    tags=["유저 관리"],
)


@app.post("/create", summary="유저 생성", description="유저 생성 API")
def create(
    user: CreateUserInput,
    db: Session = Depends(get_db),
):
    return userService.create(user, db)


@app.patch("/update/{uesrId}", summary="유저 수정", description="유저 수정 API")
def update(
    uesr_id: int,
    updateUserInput: UpdateUserInput,
    db: Session = Depends(get_db),
):
    return userService.update(uesr_id, updateUserInput, db)


@app.delete("/delete", summary="유저 삭제", description="유저 정보 삭제 API")
def delete(deleteUserInput: DeleteUserInput, db: Session = Depends(get_db)):
    return userService.delete(deleteUserInput, db)


@app.post("/list", summary="유저 리스트", description="유저 리스트 API")
def reads(readUsersInput: ReadUsersInput, db: Session = Depends(get_db)):
    return userService.filter_by(readUsersInput, db)

File: rainbow/src/main/java/com/rainbow/user/controller/UserController.java
package com.rainbow.user.controller;

import com.rainbow.common.dto.ResponseDto;
import com.rainbow.user.dto.CreateUserInput;
import com.rainbow.user.entity.User;
import com.rainbow.user.dto.DeleteUserInput;
import com.rainbow.user.dto.FilterUserInput;
import com.rainbow.user.dto.UpdateUserInput;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@Tag(name = "계정 관리")
@RequestMapping("/api/users")
public interface UserController {

    @Operation(
        summary = "유저 등록",
        description = "유저 등록 요청 API"
    )
    @PostMapping("/create")
    ResponseDto<User> createUser(CreateUserInput createUserInput);

    @Operation(
        summary = "유저 수정",
        description = "유저 수정 요청 API"
    )
    @PatchMapping("/update")
    ResponseDto<User> updateUser(Long userId, UpdateUserInput updateUserInput);

    @Operation(
        summary = "유저 리스트 조회",
        description = "유저 리스트 조회 요청 API"
    )
    @GetMapping("/list")
    ResponseDto<List<User>> findUserByFilter(FilterUserInput filterUserInput);


    @Operation(
        summary = "유저 삭제",
        description = "유저 삭제 요청 API"
    )
    @DeleteMapping("/delete")
    ResponseDto<User> deleteUser(DeleteUserInput deleteUserInput);
}
